.github/workflows/deploy.yml

```diff
@@ -0,0 +1,81 @@
+name: deploy
+
+on:
+  workflow_dispatch:
+    inputs:
+      environment:
+        description: "Target environment for deployment"
+        required: true
+        type: string
+      version:
+        description: "Version to be assigned to the built image"
+        required: true
+        type: string
+      cluster_name:
+        description: "Name of the cluster on which to perform deployment"
+        default: mangata-dev
+        required: false
+        type: string
+  workflow_call:
+    inputs:
+      environment:
+        description: "Target environment for deployment"
+        required: true
+        type: string
+      version:
+        description: "Version to be assigned to the built image"
+        required: true
+        type: string
+      cluster_name:
+        description: "Name of the cluster on which to perform deployment"
+        default: mangata-dev
+        required: false
+        type: string
+
+permissions:
+  contents: read
+  id-token: write
+  deployments: write
+
+jobs:
+  deploy:
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v3
+      - uses: jkroepke/setup-vals@v1.0.4
+      - uses: google-github-actions/auth@v1
+        with:
+          workload_identity_provider: ${{ secrets.GCP_WORKLOAD_IDENTITY_PROVIDER }}
+          service_account: ${{ secrets.GCP_SERVICE_ACCOUNT }}
+      - name: Set up GKE credentials
+        uses: google-github-actions/get-gke-credentials@v0.8.0
+        with:
+          cluster_name: ${{ inputs.cluster_name }}
+          location: europe-west1
+      
+      - name: Create a GitHub Deployment
```

Comment:
One thing I would refactor is to move the steps into separate reusable actions. This way, we will be able to create our reusable actions across all repositories in a single repository. I think it's better to do this now, and we can decide on moving them to a separate repository later.

Check this repo for inspiration: https://github.com/mangata-finance/mangata-stash/tree/main/.github/actions